app/models/edition.rb

```diff
@@ -30,6 +30,9 @@ class Edition < ActiveRecord::Base
   has_many :classification_featurings, inverse_of: :edition
   has_many :links_reports, as: :link_reportable
 
+  has_many :dependencies, class_name: 'EditionDependency', foreign_key: :edition_id, dependent: :destroy
```

Comment:
OK, I've just realised that my suggestion is utter nonsense and wouldn't work. All the same, it would be really nice if we were able to call the join table what it is (`edition_dependencies`) and come up with an alternative name for the depended upon thing. How about:
```ruby
class Edition
  has_many :edition_dependencies, dependent: :destroy
  has_many :depended_upon_contacts, through: :edition_dependencies, source: :dependable, source_type: 'Contact'
  has_many :depended_upon_editions, through: :edition_dependencies, source: :dependable, source_type: 'Edition'
end
```